index.html

```diff
@@ -572,31 +572,44 @@ <h2>
 
 </pre>
       <p>
-        This example asks the user to manually rotate the device if the Screen
-        Orientation API is not available.
+        In this example for a mobile device held in portrait, if the Screen
+        Orientation API is not supported, or the document is not fullscreen and
+        the screen orientation lock rejects, the user is alerted to rotate
+        their screen manually to landscape.
       </p>
-      <pre class='example js'>
+      <pre class='example html'>
 &lt;script&gt;
-  var start = function() {
-    screen.orientation.lock('landscape-primary').then(
-      startInternal,
-      function() {
-        alert('To start, rotate your screen to landscape.');
+function manualRotation() {
+  alert("Thank you for rotating!");
+}
 
-        var orientationChangeHandler = function() {
-          if (!screen.orientation.type.startsWith('landscape')) {
-            return;
-          }
-          screen.orientation.removeEventListener('change', orientationChangeHandler);
-          startInternal();
+async function start() {
+  try {
+    await screen.orientation.lock("landscape");
+  } catch {
+    const inPortrait = window.matchMedia("(orientation: portrait)");
+    function orientationChangeListener() {
+      function detectOrientation(e) {
+        if (e.matches) {
```

Comment:
Similarly here, it could be that other orientations will match, not just "portrait"... so this should actually be checking for "matches === landscape".